src/core/annotation.js

```diff
@@ -98,10 +98,18 @@ AnnotationFactory.prototype = /** @lends AnnotationFactory.prototype */ {
 
       case 'Widget':
         var fieldType = Util.getInheritableProperty(dict, 'FT');
-        if (isName(fieldType, 'Tx')) {
-          return new TextWidgetAnnotation(parameters);
+        fieldType = isName(fieldType) ? fieldType.name : null;
+
+        switch (fieldType) {
+          case 'Tx':
+            return new TextWidgetAnnotation(parameters);
+
+          default:
+            warn('Unimplemented widget field type "' + fieldType + '", ' +
+                 'falling back to base field type.');
+            return new WidgetAnnotation(parameters);
         }
-        return new WidgetAnnotation(parameters);
+        break;
```

Comment:
It might be a little bit cleaner to just do this instead, since it'd allow you to skip the `break` (and similar in the other file): 
```javascript
  switch (fieldType) {
    case 'Tx':
      return new TextWidgetAnnotation(parameters);
  }
  warn('Unimplemented widget field type "' + fieldType + '", ' +
       'falling back to base field type.');
  return new WidgetAnnotation(parameters);
```